edit/edit.js

```diff
@@ -61,6 +70,34 @@ baseInit.ready.then(async () => {
     '/edit/drafts',
     '/edit/global-search',
   ]);
+}).then(() => {
+  // Set up mini-header on scroll
+  const {isUsercss} = editor;
+  const el = $create('#header-sticker');
+  const scroller = isUsercss ? $('.CodeMirror-scroll') : document.body;
+  const xo = new IntersectionObserver(onScrolled, {root: isUsercss ? scroller : document});
+  scroller.appendChild(el);
+  onCompactToggled(baseInit.mqCompact);
+  baseInit.mqCompact.on('change', onCompactToggled);
+
+  /** @param {MediaQueryList} mq */
+  function onCompactToggled(mq) {
+    for (const el of $$('details[data-pref]')) {
+      el.open = mq.matches ? false : prefs.get(el.dataset.pref);
+    }
+    if (mq.matches) {
+      xo.observe(el);
+    } else {
+      xo.disconnect();
+    }
+  }
+  /** @param {IntersectionObserverEntry[]} entries */
+  function onScrolled([e]) {
```

Comment:
I think there is no guarantee that the queue will only contain a single entry. Even we only observe one target with one threshold, it is still better to process the entire queue.

Reply:
I don't think it can happen in practice, but I've switched to the last element just in case anyway. I don't see the need to process the entire queue.